### ISSUE TITLE
False positive on CanBeNonNullable: method with multiple nullables

### ISSUE THREAD

**Issue by gobetti** [detekt/detekt#7341]
## Expected Behavior
No violation should be reported.

## Observed Behavior
Violation is reported: `CanBeNonNullable - [The nullable parameter 'primaryButtonText' can be made non-nullable.]`

## Steps to Reproduce
```
@Composable
private fun ButtonsColumn(
  onPrimaryButtonClick: () -> Unit,
  primaryButtonText: String?,
  onSecondaryButtonClick: () -> Unit,
  secondaryButtonText: String?
) {
  if(primaryButtonText != null || secondaryButtonText != null) {
    Column {
      primaryButtonText?.let {
        TextButton(onClick = onPrimaryButtonClick) {
          Text(primaryButtonText)
        }
      }

      secondaryButtonText?.let {
        TextButton(onClick = onSecondaryButtonClick) {
          Text(secondaryButtonText)
        }
      }
    }
  }
}
```

## Context
In the code above, `ButtonsColumn` is a reusable component. It has more customization than shown in the code above, such that (trust me) it makes sense not to repeat that `Column` everywhere that uses it. However, some places may only need to define `primaryButtonText`, or only `secondaryButtonText`, and each button has its own style, so it makes sense to have secondary without having primary.

The bottom line is: I have multiple nullables, where all of them are indeed optional, and the only case the method won't do anything is when all are null. The way that method is currently defined makes that convenient both for the method itself, and for the call site, or I think I'd need 3 versions of it:
- `primaryButtonText: String, secondaryButtonText: String`
- `primaryButtonText: String`
- `secondaryButtonText: String`
and each caller would need logic to check the nullability to choose the right signature. If I need more nullables, e.g. a tertiary button I think I'd get an exponential explosion of overloads.

Possibly related to https://github.com/detekt/detekt/issues/5799

## Your Environment

* Version of detekt used: 1.23.4
* Version of Gradle used (if applicable): 8.7

**Comment by 3flex** [comment]
Why not just suppress the rule here? I don't see what can be changed in detekt to address this.

**Comment by gobetti** [comment]
👍 I'm not opposed to suppressing and that's what I've done for now, but we usually give it a good research, think 5 times before suppressing and still it will get caught as a smell in code review (particularly if not linked to a detekt github issue), given how rare a false positive is. Good to know that's the only solution here, thanks @3flex

**Comment by BraisGabin** [comment]
In general have a function that does nothing when you call it is a code smell. Emphasis on "in general". There are always cases where it have sense and someone can do an informed decission to suppress it.

If you can think of a pattern where this rule is always wrong we can review it but otherwise the suppress is the way to go.
